src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/StereotypeService.cs

```diff
@@ -1,27 +1,47 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Modules;
 
 namespace OrchardCore.ContentTypes.Services
 {
     public class StereotypeService : IStereotypeService
     {
         private readonly IEnumerable<IStereotypesProvider> _providers;
+        private readonly IContentDefinitionService _contentDefinitionService;
+        private readonly ILogger<StereotypeService> _logger;
 
-        public StereotypeService(IEnumerable<IStereotypesProvider> providers)
+        public StereotypeService(
+            IEnumerable<IStereotypesProvider> providers,
+            IContentDefinitionService contentDefinitionService,
+            ILogger<StereotypeService> logger)
         {
             _providers = providers;
+            _contentDefinitionService = contentDefinitionService;
+            _logger = logger;
         }
 
         public async Task<IEnumerable<StereotypeDescription>> GetStereotypesAsync()
         {
-            var descriptions = new List<StereotypeDescription>();
+            var stereotypes = (await _providers.InvokeAsync(provider => provider.GetStereotypesAsync(), _logger))
+                .Select(stereotype => stereotype.Stereotype)
```

Comment:
@MikeAlhayek , You can't do this because I defined localized descriptors in the module, which would be lost.

Reply:
okay. I am not sure why do we need to localize the widget since it is a technical name. By I pushed one more commit to improve the logic performance.

Reply:
@MikeAlhayek let me know if everything is good now.

Reply:
@hyzx86 let me know if everything is good now.

Reply:
> okay. I am not sure why do we need to localize the widget since it is a technical name. By I pushed one more commit to improve the logic performance. 
Discussed here
https://github.com/OrchardCMS/OrchardCore/issues/16005#issuecomment-2103122140

> let me know if everything is good now.

Yeah, so far it looks fine to me.